cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -120,9 +119,21 @@ public IndexedTable(String name, Table table, Table index, byte[][] columnsToInd
     this.table = table;
     this.index = index;
     this.indexedColumns = Sets.newTreeSet(Bytes.BYTES_COMPARATOR);
-    for (byte[] col : columnsToIndex) {
-      this.indexedColumns.add(col);
+    this.hasColumnWithZero = hasZeros(columnsToIndex);
+    Collections.addAll(this.indexedColumns, columnsToIndex);
+  }
+
+  private boolean hasZeros(byte[][] columns) {
+    for (byte[] column : columns) {
```

Comment:
It would be cleaner to check whether a column contains the keyDelimiter. I am not sure whether can be changed or configured, but since it is not a static final, I assume that it is not fixed as 0. Perhaps it also makes sense to make it a static final?